packages/flutter/lib/src/material/carousel.dart

```diff
@@ -56,13 +91,44 @@ class CarouselView extends StatefulWidget {
     this.scrollDirection = Axis.horizontal,
     this.reverse = false,
     this.onTap,
-    required this.itemExtent,
+    required double this.itemExtent,
     required this.children,
-  });
+  }) : allowFullyExpand = true,
+       layoutWeights = null;
+
+/// Creates a scrollable list where the size of each child widget is dynamically
+/// determined by the provided [layoutWeights].
+///
+/// The `layoutWeights` parameter is required and defines the relative size
+/// proportions of each child widget.
+///
+/// When [allowFullyExpand] is set to `true`, each child can be expanded to its
+/// maximum size while scrolling. For example, with [layoutWeights] of `[1, 7, 1]`,
```

Comment:
Is the maximum size dictated by the maximum weight?

Reply:
Yes. Updated.